libraries/AP_HAL_ChibiOS/hwdef/HAKRC4530V2/hwdef.dat

```diff
@@ -0,0 +1,209 @@
+# PA10 TIM1_CH3 TIM1 RCININT PULLDOWN
+# PB7 TIM4_CH2 TIM4 RCININT PULLDOWN
+
+# Alt config to allow RCIN on UART
+# PA10 USART2_RX USART2 ALT(1) # пока не знаю зачем это нужно
+# PB7 USART1_RX USART1 NODMA ALT(1)
+
+# SBUS inversion control pin, active low
+PC0 USART1_RXINV OUTPUT HIGH GPIO(78) POL(0)
+
+# USART2
+PA2 USART2_TX USART2
+PA3 USART2_RX USART2
+define DEFAULT_SERIAL2_PROTOCOL SerialProtocol_RCIN
+
+# USART3 (VTX)
+PB10 USART3_TX USART3
+PB11 USART3_RX USART3
+
+# UART4
+PA0 UART4_TX UART4
+PA1 UART4_RX UART4 
+define DEFAULT_SERIAL4_PROTOCOL SerialProtocol_Tramp
+
+# UART5 (ESC Telemetry)
+PC12 UART5_TX UART5
+PD2  UART5_RX UART5
+define DEFAULT_SERIAL5_PROTOCOL SerialProtocol_ESCTelemetry
+
+# USART6
+PC6 USART6_TX USART6
+PC7 USART6_RX USART6
+
+# The pins for SWD debugging with a STlinkv2 or black-magic probe (not tested)
+# PA13 JTMS-SWDIO SWD # в документации нет
+# PA14 JTCK-SWCLK SWD # в документации нет
+
+# ADC
+PC1 BATT_VOLTAGE_SENS ADC1 SCALE(1)
+# PC2 RSSI_ADC ADC1 # в документации нет, этот на pinio
+PC3 BATT_CURRENT_SENS ADC1 SCALE(1)
+
+# PWM output. 1 - 4 on ESC header
+PA8  TIM1_CH1 TIM1 PWM(1) GPIO(50) BIDIR
+PA9  TIM1_CH2 TIM1 PWM(2) GPIO(51) BIDIR
+PC8  TIM3_CH3 TIM3 PWM(3) GPIO(52) BIDIR
+PC9  TIM3_CH4 TIM3 PWM(4) GPIO(53) BIDIR
+
+# Servo
+PA10 TIM1_CH3 TIM1 PWM(5) GPIO(54)
+PB4  TIM3_CH1 TIM3 PWM(6) GPIO(55)
+PB0  TIM8_CH2N TIM8 PWM(7) GPIO(56)
+PB1  TIM8_CH3N TIM8 PWM(8) GPIO(57)
+
+
+# GPIOs
+#PA10 PINIO1 OUTPUT GPIO(81) LOW
+#PB4  PINIO2 OUTPUT GPIO(82) LOW
+
+# Servo instead of LED
+#PB3  TIM2_CH2 TIM2 PWM(9) GPIO(58)
+
+# Board LEDs
+define AP_NOTIFY_GPIO_LED_2_ENABLED 1
+PC15 LED1 OUTPUT LOW GPIO(1)
+PC14 LED2 OUTPUT LOW GPIO(2)
+define HAL_GPIO_A_LED_PIN 1
+define HAL_GPIO_B_LED_PIN 2
+
+# External LEDs
+PB3 LED_EXT1 OUTPUT GPIO(30)
+
+# Buzzer
+PC13 BUZZER OUTPUT GPIO(80) LOW
+define HAL_BUZZER_PIN 80
+
+# Camera control
+# PB9 CAM_C OUTPUT LOW GPIO(81) # в документации нет
+# define RELAY2_PIN_DEFAULT 81
+
+# Note that this board needs PULLUP on I2C pins
+PB8 I2C1_SCL I2C1 PULLUP
+PB9 I2C1_SDA I2C1 PULLUP
+
+# SPI1 - Internal IMU
+PA5 SPI1_SCK SPI1
+PA6 SPI1_MISO SPI1
+PA7 SPI1_MOSI SPI1
+# Gyro select
+PA4 MPU6000_CS CS
+PC4 GYRO_EXT_CS CS
+
+# SPI2 - OSD
+PB13 SPI2_SCK SPI2
+PB14 SPI2_MISO SPI2
+PB15 SPI2_MOSI SPI2
+# OSD max7456
+PB12 OSD_CS CS
+
+# SPI3 - dataflash
+PC10 SPI3_SCK SPI3
+PC11 SPI3_MISO SPI3
+PB5 SPI3_MOSI SPI3
+# Dataflash M25P16
+PA15 FLASH_CS CS
+
+# SPI Device table
+SPIDEV mpu6000     SPI1 DEVID1 MPU6000_CS  MODE3 1*MHZ  4*MHZ
+SPIDEV icm42688p   SPI1 DEVID1 MPU6000_CS  MODE3 1*MHZ  8*MHZ
+SPIDEV bmi270      SPI1 DEVID1 GYRO_EXT_CS MODE3 1*MHZ  8*MHZ
+SPIDEV dataflash   SPI3 DEVID1 FLASH_CS    MODE3 32*MHZ 32*MHZ
+SPIDEV osd         SPI2 DEVID4 OSD_CS      MODE0 10*MHZ 10*MHZ
+
+# One IMU rotated in yaw
+IMU Invensense   SPI:mpu6000   ROTATION_ROLL_180_YAW_90
+IMU Invensensev3 SPI:icm42688p ROTATION_ROLL_180_YAW_90
+IMU Invensensev3 SPI:bmi270    ROTATION_ROLL_180_YAW_90
```

Comment:
bmi270 is not Invensensev3 - just include the drivers and IMUs that are on the board